CHANGELOG.md

```diff
@@ -421,6 +421,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed `LightningModule.save_hyperparameters()` when attempting to save an empty container ([#7268](https://github.com/PyTorchLightning/pytorch-lightning/pull/7268))
 
 
+- Fixed Adagrad optimizer not working with DDP/GPU ([#7277](https://github.com/PyTorchLightning/pytorch-lightning/pull/7277))
+
```

Comment:
this fix applies to any optimizer that defines tensors in the init which are required for the step

this was also an issue for us with madgrad until lazy initialization was added: https://github.com/facebookresearch/madgrad/pull/5

Reply:
Would you mind making sure it is working with `madgrad` ?

Reply:
> Would you mind making sure it is working with `madgrad` ?

Reply:
> Would you mind making sure it is working with `madgrad` ?

Reply:
now lightning works fine with madgrad since the tensor initialization is delayed until the first step, where it uses the parameter's device to avoid explicitly moving the optimizer state to gpu before training

Reply:
yeah good point, but now the changelog entry is severe. 
I thought it only had to do with the creation of the state.